lib/grandpa/network.go

```diff
@@ -225,48 +232,22 @@ func (s *Service) sendNeighbourMessage() {
 }
 
 // decodeMessage decodes a network-level consensus message into a GRANDPA VoteMessage or CommitMessage
-func decodeMessage(msg *ConsensusMessage) (m GrandpaMessage, err error) {
-	var (
-		mi interface{}
-		ok bool
-	)
-
-	switch msg.Data[0] {
-	case voteType:
-		r := &bytes.Buffer{}
-		_, _ = r.Write(msg.Data[1:])
-		vm := &VoteMessage{}
-		err = vm.Decode(r)
-		m = vm
+func decodeMessage(msg scale.VaryingDataType) (m GrandpaMessage, err error) {
+	switch val := msg.Value().(type) {
+	case VoteMessage:
+		m = &val
```

Comment:
is there a reason you changed this to no longer take a `*network.ConsensusMessage` and decode the `data` in it? it might clean up the tests too

Reply:
Yeah I was doing the decoding before then passing it in but you're right it would make more sense to do it in this method

Reply:
updated!